orm/encoding/ormfield/duration.go

```diff
@@ -1,11 +1,124 @@
 package ormfield
 
 import (
+	"fmt"
 	io "io"
 
 	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
+const (
+	DurationSecondsMin = -315576000000
+	DurationSecondsMax = 315576000000
+	DurationNanosMin   = -999999999
+	DurationNanosMax   = 999999999
+)
+
+type DurationCodec struct{}
+
+func (d DurationCodec) Encode(value protoreflect.Value, w io.Writer) error {
+	// nil case
+	if !value.IsValid() {
+		_, err := w.Write(timestampDurationNilBz)
+		return err
+	}
+
+	seconds, nanos := getDurationSecondsAndNanos(value)
+	secondsInt := seconds.Int()
+	if secondsInt < DurationSecondsMin || secondsInt > DurationSecondsMax {
+		return fmt.Errorf("duration seconds is out of range %d, must be between %d and %d", secondsInt, DurationSecondsMin, DurationSecondsMax)
+	}
+	negative := secondsInt < 0
+	secondsInt -= DurationSecondsMin
+	err := encodeSeconds(secondsInt, w)
+	if err != nil {
+		return err
+	}
+
+	nanosInt := nanos.Int()
+	if nanosInt == 0 {
+		_, err = w.Write(timestampZeroNanosBz)
+		return err
+	}
+
+	if negative {
+		if nanosInt < DurationNanosMin || nanosInt > 0 {
+			return fmt.Errorf("negative duration nanos is out of range %d, must be between %d and %d", secondsInt, DurationNanosMin, 0)
+		}
+		nanosInt = -nanosInt
+	} else {
+		if nanosInt < 0 || nanosInt > DurationNanosMax {
+			return fmt.Errorf("duration nanos is out of range %d, must be between %d and %d", secondsInt, 0, DurationNanosMax)
+		}
+	}
+
+	return encodeNanos(nanosInt, w)
+}
+
+func (d DurationCodec) Decode(r Reader) (protoreflect.Value, error) {
+	isNil, seconds, err := decodeSeconds(r)
+	if isNil || err != nil {
+		return protoreflect.Value{}, err
+	}
+
+	seconds += DurationSecondsMin
```

Comment:
Same as above, a comment might be helpful here explaining why min amount is being added.